src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -216,7 +216,30 @@ StorageReplicatedMergeTree::StorageReplicatedMergeTree(
         getStorageID().getFullTableName() + " (StorageReplicatedMergeTree::mutationsFinalizingTask)", [this] { mutationsFinalizingTask(); });
 
     if (global_context.hasZooKeeper())
-        current_zookeeper = global_context.getZooKeeper();
+    {
```

Comment:
I have tested this locally,  and works great, but would like to add an integration / unit test? Still thinking of the best/right way to do that? Would be useful to have suggestions. Simplest test steps wil be:

1. Setup bad zookeeper hosts in the config file.
2. Create database and then try creating the ReplicatedMergeTreeTable.
3. Expect exception.
4. Try recreating the same table.
5. Should not throw error that the Data directory already exists for the table.

Reply:
cc: @filimonov @alexey-milovidov

Reply:
Added integration tests that test the behavior and verified that this test works fine locally.